files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharp.cs

```diff
@@ -0,0 +1,128 @@
+        var stringBuilder = new StringBuilder();
+
+        foreach(var line in lines)
+        {
+            stringBuilder.AppendLine(line);
+        }
+        stringBuilder.Length -= Environment.NewLine.Length;
+
+        return stringBuilder.ToString();
+    }
+
+    public string UseStreamReaderReadLine()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+        var stringBuilder = new StringBuilder();
+
+        while (streamReader.ReadLine() is { } fileLine)
+        {
+            stringBuilder.AppendLine(fileLine);
+        }
+        stringBuilder.Length -= Environment.NewLine.Length;
+
+        return stringBuilder.ToString();
+    }
+
+    public string UseStreamReaderReadToEnd()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+
+        return streamReader.ReadToEnd();
+    }
+
+    public string UseStreamReaderReadBlock()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+        var buffer = new char[4096];
+        int numberRead;
+        var stringBuilder = new StringBuilder();
+
+        while ((numberRead = streamReader.ReadBlock(buffer, 0, buffer.Length)) > 0)
+        {
+            stringBuilder.Append(buffer, 0, numberRead);
+        }
+
+        return stringBuilder.ToString();
+    }
+
+    public string UseStreamReaderReadBlockWithSpan()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+        var buffer = new char[4096].AsSpan();
+        int numberRead;
+        var stringBuilder = new StringBuilder();
+
+        while ((numberRead = streamReader.ReadBlock(buffer)) > 0)
+        {
+            stringBuilder.Append(buffer[..numberRead]);
+        }
+
+        return stringBuilder.ToString();
+    }
+
+    public string UseBufferedStreamObject()
+    {
+        var stringBuilder = new StringBuilder();
+
+        using var fileStream = new FileStream(_sampleFilePath, FileMode.Open, FileAccess.Read, FileShare.Read);
+        using var bufferedStream = new BufferedStream(fileStream);
+        using var streamReader = new StreamReader(bufferedStream);
+
+        while (streamReader.ReadLine() is { } fileLine)
+        {
+            stringBuilder.AppendLine(fileLine);
+        }
+        stringBuilder.Length -= Environment.NewLine.Length;
+
+        return stringBuilder.ToString();
+    }
+
+    public string UseBufferedStreamObjectWithNoFileStreamBuffer()
+    {
+        var stringBuilder = new StringBuilder();
+
+        using var fileStream = new FileStream(_sampleFilePath, FileMode.Open, FileAccess.Read, FileShare.Read, 0);
+        using var bufferedStream = new BufferedStream(fileStream);
+        using var streamReader = new StreamReader(bufferedStream);
+
+        while (streamReader.ReadLine() is { } fileLine)
+        {
+            stringBuilder.AppendLine(fileLine);
+        }
+        stringBuilder.Length -= Environment.NewLine.Length;
+
+        return stringBuilder.ToString();
+    }
```

Comment:
I think this code is silghtly different than the one in the article, can you please update the article and make sure all the article in the code matches the code in the repo?